reproman/support/jobs/orchestrators.py

```diff
@@ -931,11 +942,12 @@ def fetch(self, on_remote_finish=None):
         ref = self.job_refname
         lgr.info("Updating local dataset with changes from '%s'",
                  resource_name)
-        self.ds.repo.fetch(resource_name, "{0}:{0}".format(ref),
-                           recurse_submodules="no")
+        repo = self.ds.repo
+        repo.fetch(resource_name, "{0}:{0}".format(ref),
+                   recurse_submodules="no")
         failure = False
         try:
-            self.ds.repo.merge(ref)
+            repo.merge(ref)
```

Comment:
note to myself : In DataLad we seems to not overload `merge` in AnnexRepo, and GitRepo.merge seems to not call `.localsync`.  

question: so it is the `ds.update` call below in `else:` which will take care about merging data availability?

Reply:
Yes, the `merge` is just a merge of the jobid ref.  cdce12249 (ENH: orchestrators(datalad-pair): Do a more targeted update, 2020-04-09) provides more rationale for its existence.